packages/js/components/src/experimental-select-control/test/use-async-filter.spec.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * External dependencies
+ */
+import { act, renderHook } from '@testing-library/react-hooks';
+import { useDebounce } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { useAsyncFilter } from '../';
+
+jest.mock( '@wordpress/compose', () => ( {
+	...jest.requireActual( '@wordpress/compose' ),
+	useDebounce: jest.fn( ( cb: CallableFunction ) => cb ),
+} ) );
+
+describe( 'useAsyncFilter', () => {
+	const filter = jest.fn();
+	const onFilterStart = jest.fn();
+	const onFilterEnd = jest.fn();
+	const onFilterError = jest.fn();
+
+	afterEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	it( 'should filter the items successfully', async () => {
+		const filteredItems: string[] = [];
+		filter.mockResolvedValue( filteredItems );
+
+		const { result } = renderHook( () =>
+			useAsyncFilter( {
+				filter,
+			} )
+		);
+
+		const inputValue = '';
+
+		await act( async () => {
+			if ( result.current.onInputChange )
+				result.current.onInputChange( inputValue, {} );
+		} );
+
+		expect( useDebounce ).toHaveBeenCalledWith(
+			expect.any( Function ),
+			250
+		);
```

Comment:
Should we test the specific implementation detail? If the debouncing implementation changed to not use `useDebounce`, the test would fail, even though it might still be debouncing correctly.

Reply:
Here I'm not testing the execution/functionality of the `useDebounce` hook, it's just if it was called with the right set of params like the delay time. 
In my opinion we don't need to test external function in unit tests that's why we mocked them. So if we decide to use a diff implementation it's ok that the test fail so we can fix it with the new implementation.
But if you think `useDebounce` should be tested transitively then this is not an unit test and it is more like an integration test. Should I write integration tests instead?

Reply:
I don't think we should test the functionality of the `useDebounce` hook. But, this expect is really just ensuring that the implementation used `useDebounce`. I guess it's okay... ideally I'd like to test the debouncing. I tried something similar in the tests for `<DateTimePickerControl>`, but I had to skip the test, so perhaps it isn't really easily testable (or, I just don't know how to do it!)...

https://github.com/woocommerce/woocommerce/blob/0a039a1f1315cd4a212957cac2bc29771bbba234/packages/js/components/src/date-time-picker-control/test/index.tsx#L582

So, we could just keep the expect as is for now... it would be clear why the test fails if we were to change the implementation to not use `useDebounce`, so not a big deal.